structure_tests/structure_test_v1.go

```diff
@@ -127,18 +137,48 @@ func ProcessCommand(t *testing.T, fullCommand []string, checkOutput bool) (strin
 	if stderr != "" {
 		t.Logf("stderr: %s", stderr)
 	}
+	var exitCode int
 	if err != nil {
 		if checkOutput {
 			// The test might be designed to run a command that exits with an error.
 			t.Logf("Error running command: %s. Continuing.", err)
 		} else {
 			t.Fatalf("Error running setup/teardown command: %s.", err)
 		}
+		exitCode = err.(*exec.ExitError).Sys().(syscall.WaitStatus).ExitStatus()
+	} else {
+		exitCode = cmd.ProcessState.Sys().(syscall.WaitStatus).ExitStatus()
+	}
+	ResetEnvVars(t, originalVars)
+	return stdout, stderr, exitCode
+}
+
+func SetEnvVars(t *testing.T, vars []EnvVar) []EnvVar {
+	var originalVars []EnvVar
+	for _, env_var := range vars {
+		originalVars = append(originalVars, EnvVar{env_var.Key, os.Getenv(env_var.Key)})
+		if err := os.Setenv(env_var.Key, os.ExpandEnv(env_var.Value)); err != nil {
+			t.Fatalf("error setting env var: %s", err)
+		}
 	}
-	return stdout, stderr
+	return originalVars
 }
 
-func CheckOutput(t *testing.T, tt CommandTestv1, stdout string, stderr string) {
+func ResetEnvVars(t *testing.T, vars []EnvVar) {
```

Comment:
Ah, I think I see the line that adds the original value of all env cars that get set now. Could you add a comment that explains how it works?